src/main/java/io/beanmapper/core/BeanMatchStore.java

```diff
@@ -286,13 +297,43 @@ private BeanPropertyWrapper dealWithBeanProperty(
     }
 
     private String getBeanPropertyName(io.beanmapper.annotations.BeanProperty annotation) {
-        String beanPropertyName = annotation.value();
+        return annotation.value().isBlank() ? annotation.name() : annotation.value();
+    }
 
-        if (beanPropertyName.isEmpty()) {
-            beanPropertyName = annotation.name();
+    /**
+     * Detects whether a field annotated with the BeanProperty-annotation shadows an existing, accessible field.
+     *
+     * @apiNote In this context, an accessible field is any field which is either public, or exposes a public
+     * accessor-method.
+     *
+     * @param accessor The accessor that can be used to access to the value within the field.
+     * @param beanProperty The BeanProperty-annotation annotating the relevant field.
+     * @throws IntrospectionException May be thrown whenever an Exception occurs during the introspection of the
+     *                                relevant bean.
+     */
+    private void detectBeanPropertyFieldShadowing(final PropertyAccessor accessor, final io.beanmapper.annotations.BeanProperty beanProperty)
+            throws IntrospectionException {
+        var beanPropertyName = getBeanPropertyName(beanProperty);
+
+        var fields = accessor.getDeclaringClass().getDeclaredFields();
+        for (var field : fields) {
+            if (!field.getName().equals(accessor.getName()) && field.getName().equals(beanPropertyName) && (
```

Comment:
Misschien deze if aanpassen zodat deze variabelen gebruikt zodat het beter te lezen is wat hier precies gedaan wordt. Nu staat er een stream in.

Reply:
Ik heb hier een aparte method van de stream gemaakt. Ik denk zelf dat dat de intentie het beste weergeeft, maar als je dat anders ziet of een method hiervoor wat overdreven vindt, dan verander ik het naar variables.

Reply:
Methode naam is ook goed. Zolang het maar een duidelijke naam heeft. Voordeel met de methode wel is dat je er javadoc boven kan zetten zonder veel comments in de aanroepende methode te zetten